### DISCUSSION TITLE
Using generated C++ code without including the CasADi library in the main project

### DISCUSSION THREAD

**Discussion by Daedalus48** [casadi/casadi#3650] [Q&A]
Hi! 

According to the documentation on the web page it's possible to generate C or C++ code, then use it on a system where CasADi is not installed. This feature would be excellent for my application as I'm working with limited memory.

How would I go about doing that? In all the examples I have found about using code gen, CasADi is used in the file where the generated code is loaded as well.

Thanks in advance

**Comment by tmmsartor** [comment]
I think [this cpp example](https://github.com/casadi/casadi/blob/main/docs/examples/cplusplus/codegen_usage.cpp) already does a good job on providing a way to use CasADi C API from C/C++ (without depening on casadi library)

That source file does both the generation (and compilation) and the calling of the resulting shared library.
 For clarity I just splitted up that example in multiple files [here](https://github.com/tmmsartor/casadi_issues/tree/main/3650_use_codgenerated_solver_from_cpp), to show that `main.c` do not depend on any casadi file.
 When the tiny `mem.h` header can be made available in the deployment environment, one can make use of compact helpers to call generated functions, as shown in `main_with_mem.c`
